dev/tools/update_icons.dart

```diff
@@ -352,65 +364,65 @@ class _Icon {
       style = IconStyle.rounded;
-      shortId = id.replaceAll('_rounded', '');
+      shortId = id.replaceFirst('_rounded', '');
     } else if (id.endsWith('_sharp')) {
       style = IconStyle.sharp;
-      shortId = id.replaceAll('_sharp', '');
+      shortId = id.replaceFirst('_sharp', '');
     } else {
       style = IconStyle.regular;
       shortId = id;
     }
 
-    flutterId = id;
+    flutterId = shortId;
     for (final MapEntry<String, String> rewritePair in _identifierRewrites.entries) {
-      if (id.startsWith(rewritePair.key)) {
-        flutterId = id.replaceFirst(rewritePair.key, _identifierRewrites[rewritePair.key]);
+      if (shortId.startsWith(rewritePair.key)) {
+        flutterId = shortId.replaceFirst(rewritePair.key, _identifierRewrites[rewritePair.key]);
       }
     }
+
+    fullFlutterId = flutterId + style.idSuffix();
+    name = id.replaceAll('_', ' ');
   }
 
-  // e.g. 5g, 5g_outlined, 5g_rounded, 5g_sharp
-  String id;
-  // e.g. 5g
-  String shortId;
-  // e.g. five_g
-  String flutterId;
-  // e.g. IconStyle.outlined
-  IconStyle style;
-  // e.g. e547
-  String hexCodepoint;
+  String id;            // e.g. 5g, 5g_outlined, 5g_rounded, 5g_sharp
+  String shortId;       // e.g. 5g
+  String flutterId;     // e.g. five_g
+  String fullFlutterId; // e.g. five_g, five_g_outlined, five_g_rounded, five_g_sharp
+  String name;          // e.g. five g, five g outlined, five g rounded, five g sharp
+  IconStyle style;      // e.g. IconStyle.outlined
+  String hexCodepoint;  // e.g. e547
 
   String get mirroredInRTL => _iconsMirroredWhenRTL.contains(shortId) ? ', matchTextDirection: true' : '';
 
-  String get name => id.replaceAll('_', ' ');
-
   String get dartDoc =>
       '<i class="material-icons${style.htmlSuffix()} md-36">$shortId</i> &#x2014; material icon named "$name"';
 
   String get declaration =>
-      "static const IconData $flutterId = IconData(0x$hexCodepoint, fontFamily: 'MaterialIcons'$mirroredInRTL);";
+      "static const IconData $fullFlutterId = IconData(0x$hexCodepoint, fontFamily: 'MaterialIcons'$mirroredInRTL);";
 
   String get fullDeclaration => '''
 
   /// $dartDoc.
   $declaration
 ''';
 
-  static String platformAdaptiveDeclaration(String flutterId, _Icon agnosticIcon, _Icon iOSIcon) => '''
+  static String platformAdaptiveDeclaration(String fullFlutterId, _Icon agnosticIcon, _Icon iOSIcon) => '''
 
   /// Platform-adaptive icon for ${agnosticIcon.dartDoc} and ${iOSIcon.dartDoc}.;
-  IconData get $flutterId => !_isCupertino() ? Icons.${agnosticIcon.flutterId} : Icons.${iOSIcon.flutterId};
+  IconData get $fullFlutterId => !_isCupertino() ? Icons.${agnosticIcon.fullFlutterId} : Icons.${iOSIcon.fullFlutterId};
 ''';
 
+  /// Analogous to [String.compareTo]
+  int compareTo(_Icon b) {
+    // Sort regular icons before their variants.
+    if (flutterId == b.flutterId && style == IconStyle.regular)
+      return -1;
+    return flutterId.compareTo(b.flutterId);
+  }
```

Comment:
would it be possible to change the ordering in a separate PR? That way reviewing the PR would be much easier since we could be sure that we were only seeing reorders or only seeing new icons, not a combination.

Reply:
sure!